packages/cli/src/commands/devicepreferences/create.ts

```diff
@@ -0,0 +1,167 @@
+import inquirer from 'inquirer'
+
+import { DevicePreferenceCreate, PreferenceType } from '@smartthings/core-sdk'
+
+import { APICommand, askForInteger, askForNumber, askForRequiredString, askForString, inputAndOutputItem, userInputProcessor } from '@smartthings/cli-lib'
+
+import { tableFieldDefinitions } from '../devicepreferences'
+
+
+export default class DevicePreferencesCreateCommand extends APICommand {
+	static description = 'create a device preference'
+
+	static flags = {
+		...APICommand.flags,
+		...inputAndOutputItem.flags,
+	}
+
+	static aliases = ['device-preferences:create']
+
+	static examples = [
+		'$ smartthings devicepreferences:create                              # create a new device profile by answering questions',
+		'$ smartthings devicepreferences:create -d                           # generate a device profile by answering questions but do not actually create it',
+		'$ smartthings devicepreferences:create -i dp.json                   # create a new device profile defined by the file dp.json',
+		'$ smartthings devicepreferences:create -i dp.json -o dp-saved.json  # create a new device profile defined by the file dp.json and write the results to dp-saved.json',
+	]
+
+	async run(): Promise<void> {
+		const { args, argv, flags } = this.parse(DevicePreferencesCreateCommand)
+		await super.setup(args, argv, flags)
+
+		await inputAndOutputItem(this, { tableFieldDefinitions },
+			(_, input: DevicePreferenceCreate) => this.client.devicePreferences.create(input),
+			userInputProcessor(this))
+	}
+
+	async getInputFromUser(): Promise<DevicePreferenceCreate> {
+		const name = await askForString('Preference name:',
+			input => !input || input.match(/^[a-z][a-zA-Z0-9]{2,23}$/) ? true : 'must be camel case starting with a lowercase letter and 3-24 characters')
```

Comment:
We could write `must be camelCase. . .` in the prompt to be extra clear.

Reply:
Fixed.